public/app/core/components/editors/DashboardPicker.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import debounce from 'debounce-promise';
+import { SelectableValue, StandardEditorProps } from '@grafana/data';
+import { DashboardSearchHit } from 'app/features/search/types';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { AsyncSelect } from '@grafana/ui';
+
+export interface DashboardPickerOptions {
+  placeholder?: string;
+  isClearable?: boolean;
+}
+
+const getDashboards = (query = '') => {
+  return backendSrv.search({ type: 'dash-db', query }).then((result: DashboardSearchHit[]) => {
+    return result.map((item: DashboardSearchHit) => ({
+      uid: item.uid,
+      value: item.uid,
+      label: `${item?.folderTitle ?? 'General'}/${item.title}`,
+    }));
+  });
+};
```

Comment:
This function is used in both dashboard pickers (the only difference is extra `id` prop in legacy picker), so maybe it could be exported from one of them instead?

Reply:
> the only difference is extra id prop in legacy picker

The difference is the *type* of value -- in legacy, it is a number (id) and in the new one it is a string (uid).  I could not come up with any clever way to use the same structure ;( and wanted to just remove the id flavor